tensorflow/lite/kernels/internal/optimized/integer_ops/exp.h

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_KERNELS_INTERNAL_OPTIMIZED_INTEGER_OPS_EXP_H_
+#define TENSORFLOW_LITE_KERNELS_INTERNAL_OPTIMIZED_INTEGER_OPS_EXP_H_
+
+#include "public/gemmlowp.h"
+#include "tensorflow/lite/kernels/internal/common.h"
+#include "tensorflow/lite/kernels/internal/types.h"
+
+namespace tflite {
+namespace optimized_integer_ops {
+
+inline void QUIntExp(const ExpParams& params, const RuntimeShape& input_shape,
+                     const uint8* input_data, const RuntimeShape& output_shape,
+                     uint8* output_data) {
```

Comment:
I looked closely, but I don't see how this implementation differs materially from the int8 case. In both cases, you case to `int32` and then subtract the zero point. As far as I can tell, you could have one implementation and template on the input type. For the clamping, here you use 0 and 255, but in the int8 case, you use `std::numeric_limits<int8_t>::min()`, `std::numeric_limits<int8_t>::max()`. Can't you use std::numeric_limits::min/max for both? So, to be clear, I'm advocating for this to live as a function template in a file called `exp.h` in `internal/reference/integer_ops/`

Reply:
@talumbau , i have accepted the comments and updated the code as per your suggestion.

Regards
Amit